place_tracker/lib/place_details.dart

```diff
@@ -0,0 +1,215 @@
+            color: star1Color,
+            onPressed: () {
+              _updateStars(1);
+            },
+          ),
+          IconButton(
+            icon: Icon(Icons.star),
+            iconSize: iconSize,
+            color: star2Color,
+            onPressed: () {
+              _updateStars(2);
+            },
+          ),
+          IconButton(
+            icon: Icon(Icons.star),
+            iconSize: iconSize,
+            color: star3Color,
+            onPressed: () {
+              _updateStars(3);
+            },
+          ),
+          IconButton(
+            icon: Icon(Icons.star),
+            iconSize: iconSize,
+            color: star4Color,
+            onPressed: () {
+              _updateStars(4);
+            },
+          ),
+          IconButton(
+            icon: Icon(Icons.star),
+            iconSize: iconSize,
+            color: star5Color,
+            onPressed: () {
+              _updateStars(5);
+            },
+          ),
+        ],
+      ),
+    );
+  }
+
+  Widget _map() {
+    return Card(
+      margin: EdgeInsets.symmetric(vertical: 16.0),
+      elevation: 4.0,
+      child: Container(
+        width: 340.0,
+        height: 240.0,
+        child: GoogleMap(
+          onMapCreated: onMapCreated,
+          options: GoogleMapOptions(
+            cameraPosition: CameraPosition(
+              target: _place.latLng,
+              zoom: 16.0,
+            ),
+            zoomGesturesEnabled: false,
+            rotateGesturesEnabled: false,
+            tiltGesturesEnabled: false,
+            scrollGesturesEnabled: false,
+          ),
+        ),
+      ),
+    );
+  }
+
+  Widget _nameTextField() {
+    return Padding(
+      padding: EdgeInsets.fromLTRB(0.0, 0.0, 0.0, 16.0),
+      child: TextField(
+        decoration: InputDecoration(
+          labelText: 'Name',
+          labelStyle: TextStyle(fontSize: 18.0),
+        ),
+        style: TextStyle(fontSize: 20.0, color: Colors.black87),
+        autocorrect: false,
+        controller: _nameController,
+        onChanged: (text) {
+          setState(() {
+            _place.name = text;
+          });
+        },
+      ),
+    );
+  }
+
+  Widget _descriptionTextField() {
+    return Padding(
+      padding: EdgeInsets.fromLTRB(0.0, 0.0, 0.0, 16.0),
+      child: TextField(
+        decoration: InputDecoration(
+          labelText: 'Description',
+          labelStyle: TextStyle(fontSize: 18.0),
+        ),
+        style: TextStyle(fontSize: 20.0, color: Colors.black87),
+        maxLines: null,
+        autocorrect: true,
+        controller: _descriptionController,
+        onChanged: (text) {
+          setState(() {
+            _place.description = text;
+          });
+        },
+      ),
+    );
+  }
+
+  Widget _detailsBody() {
+    return ListView(
+      padding: EdgeInsets.fromLTRB(24.0, 12.0, 24.0, 12.0),
+      children: <Widget>[
+        _nameTextField(),
+        _descriptionTextField(),
+        _starBar(),
```

Comment:
Build a _StarBar here instead (likewise for the other TBD widgets) which creates a new Place and notifies the parent widget:

```
_StarBar(
  rating: widget.place.starRating,
  onChanged: (int value) {
    onChanged(place.copyWith(starRating: value));
  }
)
```
The parent PlaceMap would create its PlaceDetails child like this:
```
PlaceDetails(
 place: _places[marker],
 onChanged: (Place value) {
   _updatePlaceAndMarker(returnPlace, marker);
 },
)
```

Reply:
Done with slight modification. See earlier comment.